mala/datahandling/data_handler.py

```diff
@@ -515,78 +525,67 @@ def __check_snapshots(self):
 
         # Now we need to confirm that the snapshot list has some inner
         # consistency.
         if self.parameters.data_splitting_type == "by_snapshot":
             snapshot: Snapshot
+            # As we are not actually interested in the number of snapshots,
+            # but in the number of datasets, we also need to multiply by that.
             for snapshot in self.parameters.snapshot_directories_list:
                 if snapshot.snapshot_function == "tr":
                     self.nr_training_snapshots += 1
+                    self.nr_training_data += snapshot.grid_size
                 elif snapshot.snapshot_function == "te":
                     self.nr_test_snapshots += 1
+                    self.nr_test_data += snapshot.grid_size
                 elif snapshot.snapshot_function == "va":
                     self.nr_validation_snapshots += 1
+                    self.nr_validation_data += snapshot.grid_size
                 else:
                     raise Exception("Unknown option for snapshot splitting "
                                     "selected.")
 
-            # Now we need to check whether this input is believable.
+            # Now we need to check whether or not this input is believable.
             nr_of_snapshots = len(self.parameters.snapshot_directories_list)
             if nr_of_snapshots != (self.nr_training_snapshots +
                                    self.nr_test_snapshots +
                                    self.nr_validation_snapshots):
                 raise Exception("Cannot split snapshots with specified "
                                 "splitting scheme, "
                                 "too few or too many options selected")
+            if self.nr_training_snapshots == 0 and self.nr_test_snapshots == 0:
+                raise Exception("No training snapshots provided.")
+            if self.nr_validation_snapshots == 0 and self.nr_test_snapshots == 0:
+                raise Exception("No validation snapshots provided.")
+            if self.nr_training_snapshots == 0 and self.nr_test_snapshots != 0:
+                printout("DataHandler prepared for inference. No training "
+                         "possible with this setup. "
+                         "If this is not what you wanted, please revise the "
+                         "input script.", min_verbosity=0)
+                if self.nr_validation_snapshots != 0:
+                    printout("As this DataHandler can only be used for "
+                             "inference, the validation data you have "
+                             "provided will be ignored.", min_verbosity=1)
+            if self.nr_test_snapshots == 0:
+                printout("Running MALA without test data. If this is not "
+                         "what you wanted, "
+                         "please revise the input script.", min_verbosity=0)
 
-            # MALA can either be run in training or test-only mode.
```

Comment:
I simplified this comparison in a recent PR, I think it got reverted by accident in this PR.

Reply:
Should be in sync with dev now!